errai-jpa/errai-jpa-client/src/main/java/org/jboss/errai/jpa/rebind/ErraiEntityManagerGenerator.java

```diff
@@ -241,18 +241,23 @@ public static EntityManagerFactory createHibernateEntityManagerFactory(TreeLogge
         new ErraiPersistenceUnitInfo(managedTypeNames), properties);
   }
 
-  private String generateErraiEntityType(final EntityType<?> et, MethodBlockBuilder<?> pmm, List<MetaClass> globalListeners) {
+  private String generateErraiEntityType(final EntityType<?> et, MethodBlockBuilder<?> pmm, List<MetaClass> globalListeners, ClassStructureBuilder<?> classBuilder) {
     MetaClass met = MetaClassFactory.get(et.getJavaType());
     pmm.append(Stmt.codeComment(
         "**\n" +
         "** EntityType for " + et.getJavaType().getName() + "\n" +
         "**"));
     String entityTypeVarName = entitySnapshotVarName(et.getJavaType());
+    
+    String entityTypeMethodName = "createEntityType"+et.getJavaType().getName().replace('.', '_');
```

Comment:
The name of this new method should prob. be "createEntityType_" + .... for easier readability: createEntityTypeorg_jboss_errai_demo_client_shared_UserComplaint() vs. createEntityType_org_jboss_errai_demo_client_shared_UserComplaint()